File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int currSum = 0,i = 0;
        for (i = 0;i < k;i++)
        currSum += nums[i];

        int maxSum = currSum;

        for(i=k;i<nums.length;i++){
            currSum = currSum + nums[i] - nums[i-k];
            maxSum = Math.max(currSum, maxSum);
        }
        return (double)maxSum/k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currSum = 0
        for i in range(0,k):
            currSum += nums[i]

        maxSum = currSum

        for i in range(k,len(nums)):
            currSum = currSum + nums[i] - nums[i-k]
            maxSum = max(currSum,maxSum)
        return maxSum/k
        